darts/timeseries.py

```diff
@@ -37,13 +38,30 @@ def __init__(self,
         fill_missing_dates
             Optionally, a boolean value indicating whether to fill missing dates with NaN values
             in case the frequency of `series` cannot be inferred.
+        columns
+            The columns name in the same order as in the underlying DataFrame. If columns name are not unique they will
+            be replaced by a `pd.RangeIndex` converted to str starting from '0'.
         """
 
         raise_if_not(isinstance(df, pd.DataFrame), "Data must be provided in form of a pandas.DataFrame instance",
                      logger)
 
         # consistent column names
-        df.columns = range(df.shape[1])
+        def columns_are_unique(columns: Union[List[str], str]) -> bool:
+            """True if columns are unique, False if duplicates."""
+            if isinstance(columns, str):
+                return True
+            else:
+                return len(set(columns)) == len(columns)
+
+        if columns is not None and columns_are_unique(columns):
+            if isinstance(columns, str):
+                columns = [columns]
+            raise_if_not(len(columns) == df.shape[1], "`{}` columns specified when `{}` was expected from `df` "
+                                                      "shape".format(len(columns), df.shape[1]))
+            df.columns = columns
```

Comment:
Here and in line 64 we are changing the DataFrame provided by the user. I think we should not touch it, but rather do everything on the "internal" dataframe `df_`, which is a copy of `df` created on line 74.

Reply:
Ah indeed I agree

Reply:
btw relying on sort_index() to create a copy of the df might make it a bit unclear that it is a copy

Reply:
opinions @TheMP @pennfranc @Droxef ?

Reply:
Yes actually this made me notice a problem we currently have I think. In line 62 you change the columns of the original data frame right? This means that the original data frame that the user provides gets changed.

Reply:
@pennfranc  You are right I modified this behavior in the latest commits, my question was more about using ?? 
`sort_index()` to return a new DF I was personally not sure it was returning a copy or a view in that case.

Reply:
Yes good point, I think a comment clarifying at which point we are creating a copy of the df might be nice.

Reply:
+1 for a comment to make it clear. An option could also be to split it in using both `df.copy()` first, and then `df_.sort_index(inplace=True)`.